src/services/pool-creator/pool-creator.service.ts

```diff
@@ -0,0 +1,167 @@
+import {
+  Vault__factory,
+  WeightedPoolFactory__factory,
+  WeightedPool__factory
+} from '@balancer-labs/typechain';
+import { Contract } from '@ethersproject/contracts';
+import { TransactionResponse, TransactionReceipt, Web3Provider } from '@ethersproject/providers';
+import { configService } from '../config/config.service';
+import BigNumber from 'bignumber.js';
+import { BigNumber as EPBigNumber } from '@ethersproject/bignumber';
+import { sendTransaction } from '@/lib/utils/balancer/web3';
+import { defaultAbiCoder } from '@ethersproject/abi';
+import { TokenWeight } from '@/composables/pools/usePoolCreation';
+import { toNormalizedWeights } from '@balancer-labs/balancer-js';
+
+type Address = string;
+
+export interface CreatePoolReturn {
+  id: string;
+  address: Address;
+}
+
+const JOIN_KIND_INIT = 0;
+
+export interface JoinPoolRequest {
+  assets: Address[];
+  maxAmountsIn: string[];
+  userData: any;
+  fromInternalBalance: boolean;
+}
+
+export class PoolCreator {
+  public async createWeightedPool(
+    provider: Web3Provider,
+    name: string,
+    symbol: string,
+    swapFee: string,
+    tokens: TokenWeight[],
+    owner: Address
+  ): Promise<CreatePoolReturn> {
+    const weightedPoolFactoryAddress =
+      configService.network.addresses.weightedPoolFactory;
+
+    // Tokens must be sorted in order of addresses
+    tokens = this.sortTokens(tokens);
+
+    const tokenAddresses: Address[] = tokens.map((token: TokenWeight) => {
```

Comment:
This makes sense, it was re-used from the ui code @1saf what do you think about renaming this to `NewPoolToken` everywhere? And lists can be `tokens` instead of `tokenWeights`.

Reply:
Yeah sounds good, I commented the same thing on Ansaf's PR too so he's aware. Not sure who's responsible for changing.